packages/trpc/server/routers/loggedInViewer/deleteCredential.handler.ts

```diff
@@ -372,4 +340,46 @@ export const deleteCredentialHandler = async ({ ctx, input }: DeleteCredentialOp
       id: id,
     },
   });
+
+  // Backwards compatibility. Selected calendars cascade on delete when deleting a credential
+  // If it's a calendar remove it from the SelectedCalendars
+  if (credential.app?.categories.includes(AppCategories.calendar)) {
+    const selectedCalendars = await prisma.selectedCalendar.findMany({
+      where: {
+        userId: user.id,
+        integration: credential.type as string,
+      },
+    });
+
+    if (selectedCalendars.length) {
+      const calendar = await getCalendar(credential);
+
+      const calendars = await calendar?.listCalendars();
```

Comment:
Backwards compatibility for selected calendars. Eventually, once all calendars are connected to a credential, we won't have to make an API call to a 3rd party to list all calendars.